src/superqt/switch/_toggle_switch.py

```diff
@@ -97,49 +106,6 @@ def sizeHint(self) -> QtCore.QSize:
         width = opt.switch_width + text_size.width() + opt.margin * 2 + 8
         return QtCore.QSize(width, height)
 
-    def mousePressEvent(self, e):
-        if e.button() & Qt.MouseButton.LeftButton:
-            self.pressed.emit()
-        return super().mousePressEvent(e)
-
-    def mouseReleaseEvent(self, e):
-        if e.button() & Qt.MouseButton.LeftButton and self.rect().contains(e.pos()):
-            self.toggle()
-            self.released.emit()
-            self.clicked.emit()
-        return super().mouseReleaseEvent(e)
-
-    def isChecked(self) -> bool:
-        return self._checked
-
-    def setChecked(self, checked: bool) -> None:
-        if self._checked == checked:
-            return
-        self._set_checked_animated(checked)
-        self.toggled.emit(checked)
-
-    def toggle(self) -> None:
-        self.setChecked(not self.isChecked())
-        return None
-
-    def click(self) -> None:
-        """Simulate a click on the switch."""
-        self.pressed.emit()
-        self.toggle()
-        self.released.emit()
-        self.clicked.emit()
-        return None
-
-    def text(self) -> str:
-        """Text displayed next to the switch."""
-        return self._style_option.text or ""
-
-    def setText(self, text: str | None) -> None:
-        """Set the text displayed next to the switch."""
-        self._style_option.text = text
-        self.update()
-        return None
```

Comment:
these are all provided by QAbstractButton, and we should use the super-implementation if at all possible